src/client/datascience/interactive-common/intellisense/intellisenseProvider.ts

```diff
@@ -585,20 +698,55 @@ export class IntellisenseProvider implements IInteractiveWindowListener {
     private handleSignatureHelpRequest(request: IProvideSignatureHelpRequest) {
         const cancelSource = new CancellationTokenSource();
         this.cancellationSources.set(request.requestId, cancelSource);
-        this.postTimedResponse(
-            [this.provideSignatureHelp(request.position, request.context, request.cellId, cancelSource.token)],
-            InteractiveWindowMessages.ProvideSignatureHelpResponse,
-            (s) => {
-                if (s && s[0]) {
-                    return { signatureHelp: s[0]!, requestId: request.requestId };
-                } else {
-                    return {
-                        signatureHelp: { signatures: [], activeParameter: 0, activeSignature: 0 },
-                        requestId: request.requestId
-                    };
-                }
+
+        const getSignatureHelp = async (): Promise<monacoEditor.languages.SignatureHelp> => {
+            const jupyterSignatureHelp = this.provideJupyterSignatureHelp(
+                request.position,
+                request.cellId,
+                cancelSource.token
+            );
+
+            const lsSignatureHelp = this.provideSignatureHelp(
+                request.position,
+                request.context,
+                request.cellId,
+                cancelSource.token
+            );
+
+            // Capture telemetry for each of the two providers.
+            // Telemetry will be used to improve how we handle intellisense to improve response times for code completion.
+            // NOTE: If this code is around after a few months, telemetry isn't used, or we don't need it anymore.
+            // I.e. delete this code.
+            sendTelemetryWhenDone(Telemetry.CompletionTimeFromLS, lsSignatureHelp);
+            sendTelemetryWhenDone(Telemetry.CompletionTimeFromJupyter, jupyterSignatureHelp);
```

Comment:
seems these are for completion not signature help?

Reply:
Let me have a check

Reply:
Removed